### PR TITLE
Automated releases

### PR DESCRIPTION
**Pull request by hawk23**
## Description
Adds automation using GH actions for the whole release process.

## Changes
- Add workflow `start-release-train.yml`
    - Creates the release branch
    - Bumps the version in all necessary places
    - Does a publish dry-run: `dart pub publish --dry-run`
    - Test run: https://github.com/bitmovin/bitmovin-player-flutter/actions/runs/7287704111
- Add workflow `finish-release-train.yml`
    - Adds the release date to CHANGELOG.md
    - Pushes a version tag to trigger the actual publishing of the package (this is a requirement of `dart-lang/setup-dart/.github/workflows/publish.yml@v1`)
    - Creates a PR to merge release branch, including version tag, back to main
    - Test run: https://github.com/bitmovin/bitmovin-player-flutter/actions/runs/7287786241
- Add workflow `publish.yml`
   - Based on this guide: https://dart.dev/tools/pub/automated-publishing
   - When a version tag is pushed, publishes the package to `pub.dev`
   - Creates the according GH release
   - Generates the API doc and uploads it to Bitmovin's CDN
   - Test run: https://github.com/bitmovin/bitmovin-player-flutter/actions/runs/7287788834

Sidenote: The GH release, release branch, merge PR, and tag, that were created by the test run are deleted already again, to have a clean repo state.

## Tests
n/a

## Checklist (for PR submitters and reviewers)
- [x] 🗒 `CHANGELOG.md` entry for new/changed features, bug fixes or important code changes: n/a as only internal changes
- [x] 🧪 Tests added and/or updated: n/a
- [x] 📢 New public API is fully documented

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 
 ## [0.4.0] - 2023-12-12
 ### Added
-Support for Picture-in-Picture (PiP) playback on iOS
+- Support for Picture-in-Picture (PiP) playback on iOS
```
Comment (hawk23) [inline CHANGELOG.md:11]: Sneak fix

File: ios/bitmovin_player.podspec
Comment (hawk23) [inline ios/bitmovin_player.podspec:1]: To simplify the workflow, the version number and other info is now taken form `pubspec.yaml`. One less version to bump

File: .github/workflows/start-release-train.yml [hunk 2]
```diff
@@ -1,9 +1,80 @@
...
+          awk 'BEGIN {count=0} /## \[/ {count++; if (count == 2) exit} {print}' CHANGELOG.md
+
+      - name: Bump package version
+        run: | 
+          dart pub global activate cider
+          dart pub global run cider version ${{ inputs.version_number }}
+          flutter pub get
+
+      - name: Install pods to update Podfile.lock
+        run: (cd example/ios && pod install --repo-update)
```
Comment (rolandkakonyi) [inline .github/workflows/start-release-train.yml:60]: nit: you could use `working-directory` option:
```suggestion
        run: pod install --repo-update
        working-directory: example/ios
```

**Review by rolandkakonyi — APPROVED** [review]

File: .github/workflows/start-release-train.yml (hunk 2 above)
Comment (hawk23) [reply .github/workflows/start-release-train.yml]: Nice improvement: 3ce3db64

**Pull request merged**